### DISCUSSION TITLE
Mini browser and X-frame-options

### DISCUSSION THREAD

**Discussion by RomanNikitenko** [eclipse-theia/theia#9162] [General]
We use the mini browser related functionality in `Che` to show user's application page.
Some applications use an HTTP header `X-frame-options` with `deny` or `sameorigin` value.

The page is refused for such cases with an error like:
`Refused to display https://www.some.path in a frame because it set `X-Frame-Options` to sameorigin`.
The empty page is displayed and probably a user has no idea what's the problem.  

I wonder if there is a way to detect `X-frame-options` or catch the case when a page is refused.
We could display a warning for a user that we detected such problem and some info how it could be fixed on an application side.

I played with some hacks which I found on different resources, but, unfortunately, I haven't find a reliable solution.
Anyway if someone has an idea how we could handle such use cases - please let me know and share your thoughts.

Thanks in advance!

**Comment by wassimkha** [comment]
@RomanNikitenko were you able to find a solution or workaround for this issue? or be able to detect `X-frame-options` and warn the users?

**Comment by RomanNikitenko** [comment]
Unfortunately - no...
